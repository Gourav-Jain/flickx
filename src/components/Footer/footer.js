import React from 'react'
import { Divider, Segment, Grid, Header, Message } from 'semantic-ui-react';
import { FavoriteList } from './favlist'

export const Footer = (props) => {
    const { favourites, removeFavourite, watchLater, removeWatchLater } = props;
    return (
        <>
            <Divider />
            <Segment>
                <Grid columns={2} relaxed='very'>
                    <Grid.Column>
                        <Header as="h3" dividing textAlign='center'>
                            Your Favorites Movies
                        </Header>
                        {
                            favourites && favourites.length > 0 ? <FavoriteList items={favourites} clicked={removeFavourite} /> : <Message warning>
                                <p>No videos in your favourite list</p>
                            </Message>
                        }


                    </Grid.Column>
                    <Grid.Column>
                        <Header as="h3" dividing textAlign='center'>
                            Your Watch Later Movies
                        </Header>
                        {
                            watchLater && watchLater.length > 0 ? <FavoriteList items={watchLater} clicked={removeWatchLater} /> : <Message warning>
                                <p>No videos in your watch later list</p>
                            </Message>
                        }


                    </Grid.Column>
                </Grid>
                <Divider vertical>AND</Divider>
            </Segment>
        </>
    )
}

export default Footer

