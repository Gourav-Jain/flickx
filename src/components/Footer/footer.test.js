
import React from "react";
import { shallow } from "enzyme";

import { Footer } from "./footer";

describe("<Footer />", () => {
    const props = {
        favourites: [],
        removeFavourite: jest.fn,
        watchLater: [],
        removeWatchLater: jest.fn
    };

    beforeEach(jest.clearAllMocks);

    describe("@actions", () => {
        test("should display footer", () => {
            const footer = shallow(<Footer {...props} />);
            expect(footer).toMatchSnapshot();
        });
    });
});