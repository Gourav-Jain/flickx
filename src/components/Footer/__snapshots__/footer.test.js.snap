// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> @actions should display footer 1`] = `ShallowWrapper {}`;
